### PR TITLE
feat(tabs): Allow setting optional classes on each tab.

### PR DESCRIPTION
**Pull request by amahfouz**
Allow setting any set of optional CSS classes on a uib-tab element.

This is to allow specifying classes such as "btn-sm" for tab pills.

### CODE REVIEW

File: src/tabs/test/tabs.spec.js [hunk 1]
```diff
@@ -34,16 +34,18 @@ describe('tabs', function() {
       scope.first = '1';
       scope.second = '2';
       scope.actives = {};
+      scope.classesFirst = "first-classes-1";
```
Comment (wesleycho) [inline src/tabs/test/tabs.spec.js]: Better named `firstClass` - also `first-class-1`, and use single quotes

File: src/tabs/test/tabs.spec.js [hunk 2]
```diff
@@ -34,16 +34,18 @@ describe('tabs', function() {
...
+      scope.classesSecond = "second-classes-1 second-classes-2";
```
Comment (wesleycho) [inline src/tabs/test/tabs.spec.js]: Ditto

**Comment by icfantv** [comment]
This also needs documentation and demo updates.

**Comment by amahfouz** [comment]
I fixed all reported issues. 

Any update on the status of this?

Thanks.

**Comment by icfantv** [comment]
@amahfouz looks like you need to rebase first.

**Comment by wesleycho** [comment]
Needs squashing and rebasing (squash first).

**Comment by amahfouz** [comment]
Thanks for the prompt reply. I had already done the squash. 

Now I did:   git pull --rebase and did not get any conflicts, but there is nothing to push. What am I missing?

**Comment by wesleycho** [comment]
git pull --rebase origin master

**Comment by amahfouz** [comment]
Got it. Thanks. I have now squashed, rebased, and force pushed.

**Comment by amahfouz** [comment]
Tests now fail :( 
Still got work to do.

**Comment by amahfouz** [comment]
Checks now running. Let me know if I need to do anything else.

File: src/tabs/tabs.js [hunk 3]
```diff
@@ -132,6 +133,10 @@ angular.module('ui.bootstrap.tabs', [])
         });
       }
 
+      if (attrs.classes === undefined) {
```
Comment (RobJacobs) [inline src/tabs/tabs.js]: Angular has built in functions for checking if variables are [undefined](https://docs.angularjs.org/api/ng/function/angular.isUndefined) or are [defined](https://docs.angularjs.org/api/ng/function/angular.isDefined)

Comment (amahfouz) [inline src/tabs/tabs.js]: Do I need to fix this, resquash, rebase, etc. or this is within the margin of tolerance?

Comment (wesleycho) [inline src/tabs/tabs.js]: Use `angular.isUndefined` here

**Comment by wesleycho** [comment]
This needs to be rebased.

**Comment by amahfouz** [comment]
I have made the last requested fix, squashed, and rebased. Please review.

File: src/tabs/tabs.js [hunk 4]
```diff
@@ -143,6 +144,10 @@ angular.module('ui.bootstrap.tabs', [])
         }
       }
 
+      if (angular.isUndefined(attrs.classes)) {
+        scope.classes = "";
```
Comment (wesleycho) [inline src/tabs/tabs.js]: Use `'` instead of `"`

**Comment by amahfouz** [comment]
Done. Squashed, rebased and pushed.

**Pull request closed without merging**